cadquery/occ_impl/assembly.py

```diff
@@ -289,3 +301,121 @@ def toJSON(
+    assy_name: str,
+    glue: Optional[bool] = False,
+    tol: Optional[float] = None,
+) -> TDocStd_Document:
+    """
+    Converts the assembly to a fused compound and saves that within the document
+    to be exported in a way that preserves the face colors. Because of the use of
+    boolean operations in this method, performance may be slow in some cases.
+
+    :param assy: Assembly that is being converted to a fused compound for the document.
+    :param assy_name: Label that is applied to the top level object in the document.
+    """
+
+    # Prepare the document
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+
+    # To fuse the parts of the assembly together
+    fuse_op = BRepAlgoAPI_Fuse()
+    args = TopTools_ListOfShape()
+    tools = TopTools_ListOfShape()
+
+    # If there is only one solid, there is no reason to fuse, and it will likely cause problems anyway
+    top_level_shape = None
+
+    # Handle a top level object that was added during Assembly initialization
+    if assy.obj != None:
+        # If we do not have an arg yet, add this as the first.
+        if args.IsEmpty():
+            args.Append(cast(Shape, cast(Workplane, assy.obj).val()).wrapped)
+
+    # Handle the children of the assembly
+    for child in assy.children:
+        # If there are no args yet, add the first shape to that list, otherwise make the shapes tools
+        if args.IsEmpty():
+            for i, shape in enumerate(child.toShapeList()):
+                if i == 0:
+                    args.Append(shape.wrapped)
+                else:
+                    tools.Append(shape.wrapped)
+        else:
+            for shape in child.toShapeList():
+                tools.Append(shape.wrapped)
+
+    # If the tools are empty, it means we only had a single shape and do not need to fuse
+    if args.IsEmpty():
+        raise Exception(
+            "Error: A top level shape of assembly " + assy_name + " is not present."
+        )
+    elif tools.IsEmpty():
+        top_level_shape = args.First()
+    else:
+        # Allow the caller to configure the fuzzy and glue settings
+        if tol:
+            fuse_op.SetFuzzyValue(tol)
+        if glue:
+            fuse_op.SetGlue(BOPAlgo_GlueEnum.BOPAlgo_GlueShift)
+
+        fuse_op.SetArguments(args)
+        fuse_op.SetTools(tools)
+        fuse_op.Build()
+
+        top_level_shape = fuse_op.Shape()
+
+    # Add the fused shape as the top level object in the document
+    top_level_lbl = shape_tool.AddShape(top_level_shape, False)
+    TDataStd_Name.Set_s(top_level_lbl, TCollection_ExtendedString(assy_name))
+
+    # Walk the assembly->part->shape->face hierarchy and add subshapes for all the faces
+    for part in assy.children:
```

Comment:
Could `toShapeList` be not needed?
```suggestion
    for _, part in assy.traverse():
       for shape in part.shapes:
           for face in shape.Faces(): 
```